Jellyfin.Server.Implementations/StorageHelpers/StorageHelper.cs

```diff
@@ -0,0 +1,101 @@
+    /// </summary>
+    /// <param name="applicationPaths">The application paths.</param>
+    /// <param name="logger">Logger.</param>
+    public static void TestCommonJfPathsForStorageCapacity(IApplicationPaths applicationPaths, ILogger logger)
+    {
+        TestDataDirectorySize(applicationPaths.DataPath, logger, TwoGigabyte);
+        TestDataDirectorySize(applicationPaths.LogDirectoryPath, logger, FiveHundredAndTwelveMegaByte);
+        TestDataDirectorySize(applicationPaths.CachePath, logger, TwoGigabyte);
+        TestDataDirectorySize(applicationPaths.ProgramDataPath, logger, TwoGigabyte);
+        TestDataDirectorySize(applicationPaths.TempDirectory, logger, TwoGigabyte);
+    }
+
+    /// <summary>
+    /// Gets the free space of a specific directory.
+    /// </summary>
+    /// <param name="path">Path to a folder.</param>
+    /// <returns>The number of bytes available space.</returns>
+    public static FolderStorageInfo GetFreeSpaceOf(string path)
+    {
+        try
+        {
+            var driveInfo = new DriveInfo(path);
+            return new FolderStorageInfo()
+            {
+                Path = path,
+                FreeSpace = driveInfo.AvailableFreeSpace,
+                UsedSpace = driveInfo.TotalSize - driveInfo.AvailableFreeSpace,
+                StorageType = driveInfo.DriveType.ToString(),
+            };
+        }
+        catch
+        {
+            return new FolderStorageInfo()
+            {
+                Path = path,
+                FreeSpace = -1,
+                UsedSpace = -1,
+                StorageType = null,
+            };
+        }
+    }
+
+    private static void TestDataDirectorySize(string path, ILogger logger, long threshold = -1)
+    {
+        logger.LogDebug("Check path {TestPath} for storage capacity", path);
+        var drive = new DriveInfo(path);
+        if (threshold != -1 && drive.AvailableFreeSpace < threshold)
+        {
+            throw new InvalidOperationException($"The path `{path}` exceeds the minimum required free capacity of {BytesToString(threshold)}");
+        }
+
+        logger.LogInformation(
+            "Storage path `{TestPath}` ({StorageType}) successfully tested with {FreeSpace} left free which is over the minimum of {MinFree}.",
+            path,
+            drive.DriveType,
+            BytesToString(drive.AvailableFreeSpace),
+            BytesToString(threshold));
+    }
+
+    /// <summary>
+    /// Formats a size in bytes into a common human readable form.
+    /// </summary>
+    /// <remarks>
+    /// Taken and slightly modified from https://stackoverflow.com/a/4975942/1786007 .
+    /// </remarks>
+    /// <param name="byteCount">The size in bytes.</param>
+    /// <returns>A human readable approximate representation of the argument.</returns>
+    public static string BytesToString(long byteCount)
+    {
+        string[] suf = ["B", "KB", "MB", "GB", "TB", "PB", "EB"];
```

Comment:
this should be extracted

Reply:
In my opinion this is actually the right place for that. StorageHelper should contain all the helper methods to get data about the underlying storage system and a humanizer for FileSize does fit here. But where to otherwise?

Reply:
Sorry, I meant to take the array out of the function and make it static :)